client/src/main/scala/com/dwolla/cloudflare/RateLimitClient.scala

```diff
@@ -15,114 +13,97 @@ import io.circe.syntax._
 import io.circe.{Decoder, DecodingFailure, HCursor, Json}
 import fs2._
 import org.http4s.Method._
-import org.http4s._
 import org.http4s.circe._
 import org.http4s.client.dsl.Http4sClientDsl
 
 import scala.language.higherKinds
+import scala.util.matching.Regex
 
 trait RateLimitClient[F[_]] {
   def list(zoneId: ZoneId): Stream[F, RateLimit]
-  def getById(zoneId: ZoneId, rateLimitId: RateLimitId): Stream[F, RateLimit]
+  def getById(zoneId: ZoneId, rateLimitId: String): Stream[F, RateLimit]
   def create(zoneId: ZoneId, rateLimit: CreateRateLimit): Stream[F, RateLimit]
   def update(zoneId: ZoneId, rateLimit: RateLimit): Stream[F, RateLimit]
-  def delete(zoneId: ZoneId, rateLimitId: RateLimitId): Stream[F, RateLimitId]
+  def delete(zoneId: ZoneId, rateLimitId: String): Stream[F, RateLimitId]
+
+  def getByUri(uri: String): Stream[F, RateLimit] = parseUri(uri).fold(Stream.empty.covaryAll[F, RateLimit]) {
```

Comment:
@aksondhi @nleeper Do you think we should add `getByUri` and `parseUri` methods to our other clients too?

Reply:
How would it benefit the other clients?